### ISSUE TITLE
Create on download

### ISSUE THREAD

**Issue by Worthaboutapig** [lordfuoco/vue-xlsx#6]
I may be missing something here, but it seems, as the `download` component is a child of the other components, a workbook with sheets is created regardless. How to I create the workbook with sheets on demand? i.e. I don't want to create the workbook until it's requested by the download click?

**Comment by nmezzopera** [comment]
@Worthaboutapig True, the default use case is to have the workbook live with the data, if you do not want this behaviour, currently it's a bit of a hack, but it's a good idea to implement this, I will think of an efficient way to do so.

if you want to create the workbook only when download is pressed: 

- put `xlsx-workbook` under v-if and add a `ref` to it like `xlsx`
- on `xlsx-workbook` add a listener for `change` with a function that will 'click'  `xlsxl-download`  button ( with the ref ) 
- inside `xlsx-workbook` add `xlsx-download` with a ref
- have a download button that set the v-if to true
